optuna/study.py

```diff
@@ -523,6 +529,17 @@ def _create_record_and_aggregate_column(trial):
 
         return df
 
+    @experimental("1.4.0")
+    def stop(self) -> None:
+        """Stop the optimization after the current trial finishes."""
```

Comment:
I think this method needs a more detailed description. For example,
```suggestion

        """Exit from the current optimization loop after the running trials finish.

        This method lets the running `optimize` method return immediately after
        all trials which the `optimize` method spawned finishes.
        This method does not affect any behaviors of parallel or successive
        study processes.
        
        Raises:
            RuntimeError:
                If this method is called outside an objective function or callback.
        """
```

Reply:
I agree. This method stops only a single study process if users run multiple processes. I think this behavior fits the behavior of  `study.optimize`, but some users may expect to stop all processes. So, this description is useful for such users.
I feel it may be more helpful for users if we show how to stop all processes (e.g., setting a flag to study.user_attr). It optional and we can separate PRs.

A net. `optimize` seems to be linked to the reference:
```
:meth:`~optuna.study.Study.optimize`
```